staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -456,12 +564,120 @@ func (d *draPlugin) PublishResources(_ context.Context, resources resourceslice.
 	return nil
 }
 
-// RegistrationStatus implements [DRAPlugin.RegistrationStatus].
-func (d *draPlugin) RegistrationStatus() *registerapi.RegistrationStatus {
+// RegistrationStatus returns the result of registration, nil if none received yet.
+func (d *Helper) RegistrationStatus() *registerapi.RegistrationStatus {
 	if d.registrar == nil {
 		return nil
 	}
+	// TODO: protect against concurrency issues.
 	return d.registrar.status
 }
 
-func (d *draPlugin) internal() {}
+// serializeGRPC locks a mutex if serialization is enabled.
+// Either way it returns a method that the caller must invoke
+// via defer.
+func (d *Helper) serializeGRPC() func() {
+	if !d.serialize {
+		return func() {}
+	}
+	d.grpcMutex.Lock()
+	return d.grpcMutex.Unlock
+}
+
+// nodePluginImplementation is a thin wrapper around the helper instance.
+// It prevents polluting the public API with these implementation details.
+type nodePluginImplementation struct {
+	*Helper
+}
+
+// NodePrepareResources implements [drapb.NodePrepareResources].
+func (d *nodePluginImplementation) NodePrepareResources(ctx context.Context, req *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error) {
+	// Do slow API calls before serializing.
+	claims, err := d.getResourceClaims(ctx, req.Claims)
+	if err != nil {
+		return nil, fmt.Errorf("get resource claims: %w", err)
+	}
+
+	// Serialize if enabled, otherwise NOP.
+	defer d.serializeGRPC()()
```

Comment:
Instead of a comment each time, should we call the function `serializeGRPCIfEnabled()`?